go-controller/pkg/node/gateway_shared_intf_linux.go

```diff
@@ -133,29 +133,46 @@ func deleteLocalNodeAccessBridge() error {
 	return nil
 }
 
-func addSharedGatewayIptRules(service *kapi.Service) {
-	rules := getGatewayIPTRules(service, nil)
+// If A Service External Traffic Policy==Local and backend is a host-networked pod
```

Comment:
I think regardless of a host-network pod or not, it needs to be DNAT'ed straight to the endpoint anytime a host-> external traffic policy local type traffic occurs. If the service has a host network endpoint, then DNAT directly to that (like you do below), however if the service has an OVN pod endpoint, you need to also DNAT directly to that. The packet will be delivered via mp0, although it would get SNAT'ed once to the mp0 IP (that should be somewhat OK since it is a host sourced packet and mp0 is a host source address).

Reply:
I don't want there to be iptables rules DNATing directly to OVN managed endpoints, OVN should be responsible if DST-IP is in OVN's net.  I think the SNAT to MP0 is alright, since it's still an address on the Host.